CHANGES.md

```diff
@@ -1,5 +1,85 @@
 CHANGES.md lists intentional changes between the Strada (Typescript) and Corsa (Go) compilers.
 
+# Feature-Level Description
+
+At a high level, Javascript support in Corsa is intended to expose Typescript features in a .js file, working exactly as they do in Typescript with different syntax.
+This differs from Strada, which has many Javascript features that do not exist in Typescript at all, and quite a few differences in features that overlap.
+For example, Corsa uses the same rule for checking calls in both Typescript and Javascript; Strada lets you skip parameters with type `any`.
+And because Corsa uses the same rule for optional parameters, it fixes subtle Strada bugs with `"strict": true` in Javascript.
+
+We primarily want to support people writing modern Javascript, using things like ES modules, classes, destructuring, etc.
+Not CommonJS modules and constructor functions, although those do still work.
+However, we have trimmed a lot of unused or underused features.
```

Comment:
Can we say that we did so because it makes the implementation simpler / more robust? (We of course aren't cutting for no reason)